keystorage/aws/src/main/java/tech/pegasys/signers/aws/AwsSecretsManager.java

```diff
@@ -68,6 +81,54 @@ public Optional<String> fetchSecret(final String secretName) {
     }
   }
 
+  private ListSecretsIterable listSecrets(
+      final List<String> tagKeys, final List<String> tagValues) {
+    final ListSecretsRequest.Builder listSecretsRequestBuilder = ListSecretsRequest.builder();
+    if (!tagKeys.isEmpty()) {
+      listSecretsRequestBuilder.filters(
+          Filter.builder().key(FilterNameStringType.TAG_KEY).values(tagKeys).build());
+    }
+    if (!tagValues.isEmpty()) {
+      listSecretsRequestBuilder.filters(
+          Filter.builder().key(FilterNameStringType.TAG_VALUE).values(tagValues).build());
+    }
+    return secretsManagerClient.listSecretsPaginator(listSecretsRequestBuilder.build());
+  }
+
+  public <R> Collection<R> mapSecrets(
+      final List<String> tagKeys,
```

Comment:
Does this need to be a List? The filter takes a collection so there probably no need to this be a list, a collection is probably sufficient.